nntrainer/graph/network_graph.cpp

```diff
@@ -331,44 +340,48 @@ int NetworkGraph::addLossLayer(const LossType loss_type) {
   NN_RETURN_STATUS();
 
   addLayerNode(layer);
+  setEdge(adj.size() - 1);
+  topologicalSort();
 
   return ML_ERROR_NONE;
 }
 
 void NetworkGraph::setOutputLayers() {
 
-  if (layers.back()->getNumOutputs() > 0 &&
-      layers.back()->output_layers.size() > 0) {
+  auto &last_layer = adj.back().front().layer;
+  if (last_layer->getNumOutputs() > 0 && last_layer->output_layers.size() > 0) {
     throw std::runtime_error("last layer already has a output layer");
   }
 
-  layers.back()->setNumOutputs(1);
-  layers.back()->output_layers.clear();
-  layers.back()->output_layers.push_back("__exit__");
+  last_layer->setNumOutputs(1);
```

Comment:
I wonder if it is always 1?

Reply:
Yes it is. __exit__

Reply:
Yes, it is, the statement is just for easier reading.